Solution

TarasFirst opened the pull request.

Commit:
create and fill .gitignore

Commit:
add manage.py

Commit:
Create models

- create models Movie, Director.

Commit:
Create serializers.py

- add MovieSerializer.

Commit:
Create endpoints

- path movies/ and path movies/<int:pk>/.

Commit:
Create views

- add movie_list and movie_detail.

Commit:
Registration models in admin

- register Movie and User, UserAdmin.

Commit:
Add app files

- migrations, __init__.py, apps.py.

Commit:
Create and fill requirements.txt

Commit:
Add two apps to installed apps and add AUTH_USER_MODEL in settings.py

Commit:
Add path api/cinema/ to movie_service/urls.py

Commit:
Final commit

- add standart project files asgi, wsgi, init.

Dimosphen1 commented on .gitignore:
Include only the required entities in the `.gitignore` file, remove redundant ones

Dimosphen1 commented on cinema/models.py:
```diff
@@ -0,0 +1,15 @@
+from django.db import models
```
Add one blank line between different groups of imports here and in the other places to follow PEP-8

Dimosphen1 commented on cinema/views.py:
```diff
@@ -0,0 +1,35 @@
+    elif request.method == "DELETE":
+        movie.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+    else:
```
Check method explicitly before performing an operation, make sure that it is `GET` before returning a response

Dimosphen1 reviewed (changes requested):
Several changes were requested

Commit:
Fix .gitignore

- remove redundant entities.

Commit:
Fix with PEP-8 in imports

- add one blank line between different groups of imports.

Commit:
Fix views.py

- add a check method explicitly before performing an operation, make sure that it is GET before returning a response in movie_detail.

Commit:
Add one blank line between different groups of imports

Commit:
Final fix commit with

- add readme.

pavlopro commented on cinema/views.py:
```diff
@@ -0,0 +1,35 @@
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+    else:
```
The same with check method explicitly here as well

Commit:
Fix views.py

- add a check method for movie_list explicitly before performing an operation,
make sure that it is GET before returning a response in movie_detail.

pavlopro reviewed (approved):
GJ!
